bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/datamodel/FreeAtHomeDatapointGroup.java

```diff
@@ -0,0 +1,294 @@
+        }
+
+        return ret;
+    }
+
+    public boolean isInteger() {
+        boolean ret = false;
+
+        functionString = PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+
+        switch (functionString) {
+            case PidTranslationUtils.PID_VALUETYPE_INTEGER:
+                ret = true;
+                break;
+            default:
+                ret = false;
+                break;
+        }
+
+        return ret;
+    }
+
+    public boolean isReadOnly() {
+        boolean result = true;
+
+        if (DatapointGroupDirection.INPUTOUTPUT == datapointGroupDirection) {
+            result = false;
+        }
+
+        if (DatapointGroupDirection.INPUT == datapointGroupDirection) {
+            result = false;
+        }
+
+        return result;
+    }
+
+    public int getMax() {
+        return PidTranslationUtils.getMax(String.format("0x%04X", pairingId));
+    }
+
+    public int getMin() {
+        return PidTranslationUtils.getMin(String.format("0x%04X", pairingId));
+    }
+
+    public String getLabel() {
+        return PidTranslationUtils.getShortTextForPairingId(String.format("0x%04X", pairingId));
+    }
+
+    public String getDescription() {
+        return PidTranslationUtils.getDescriptionTextForPairingId(String.format("0x%04X", pairingId));
+    }
+
+    public String getValueType() {
+        return PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+    }
+
+    public String getTypePattern() {
+        String pattern = "";
+
+        functionString = PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+
+        switch (functionString) {
+            case PidTranslationUtils.PID_VALUETYPE_DECIMAL:
+                pattern = "%.1f";
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_INTEGER:
+                pattern = "%d";
+                break;
+            default:
+                pattern = "";
+                logger.debug("Type pattern not forund for PairingID {} - using default",
+                        String.format("0x%04X", pairingId));
+                break;
+        }
+
+        return pattern;
+    }
+
+    public ValueStateConverter getValueStateConverter() {
+        ValueStateConverter valueStateConverter = new BooleanValueStateConverter();
+
+        functionString = PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+
+        switch (functionString) {
+            case PidTranslationUtils.PID_VALUETYPE_BOOLEAN:
+                valueStateConverter = new BooleanValueStateConverter();
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_DECIMAL:
+                valueStateConverter = new DecimalValueStateConverter();
+                break;
+            case PidTranslationUtils.PID_VALUETYPE_INTEGER:
+                valueStateConverter = new DecimalValueStateConverter();
+                break;
```

Comment:
Do you actually need both? The implementations look very similar.

Reply:
This can be reduced for this and on an other place in the code. However later could be needed, therefore I still keep the "case"-es to be able to reactivate if needed later.